packages/in_app_purchase/ios/Classes/InAppPurchasePlugin.m

```diff
@@ -154,7 +154,7 @@ - (void)addPayment:(FlutterMethodCall *)call result:(FlutterResult)result {
     SKMutablePayment *payment = [SKMutablePayment paymentWithProduct:product];
     payment.applicationUsername = [paymentMap objectForKey:@"applicationUsername"];
     NSNumber *quantity = [paymentMap objectForKey:@"quantity"];
-    payment.quantity = quantity ? quantity.integerValue : 1;
+    payment.quantity = (quantity != nil) ? quantity.integerValue : 1;
```

Comment:
what is the different or benefit we gain with this change?

Reply:
It fixes a static analyzer warning:
```
    157:24: warning: Converting a pointer value of type 'NSNumber *' to a primitive boolean value; instead, either compare the pointer to nil or call -boolValue
        payment.quantity = quantity ? quantity.integerValue : 1;
```